app/controllers/api/v1/formats_controller.rb

```diff
@@ -0,0 +1,77 @@
+# coding: utf-8
+class Api::V1::FormatsController < ApiController
+
+  include Swagger::Blocks
+
+  swagger_path '/formats' do
+    operation :get do
+      key :summary, 'Get all formats'
+      key :description, 'List all known formats'
+      key :operationId, 'getFormats'
+      security do
+        key :api_token, []
+      end
+      key :tags, ['Formats']
+
+      extend Api::SwaggerResponses
+
+      success_schema(model: 'Format', format: :array)
+    end
+  end
+  def index
+    render api_response data: Format.all
+  end
+
+  swagger_schema :FormatInput do
+    allOf do
+      schema do
+        key :required, [:name, :specification]
+
+        property :name,
+                 type: :string,
+                 description: 'The name of the format'
+
+        property :specification,
+                 type: :string,
+                 description: 'A longer description that fully explains the format'
+
+      end
+    end
+  end
+
+  swagger_path '/formats' do
+    operation :post do
+      key :summary, 'create  a format'
+      key :operationId, 'createFormat'
+      security do
+        key :api_token, []
+      end
+      key :tags, ['Formats']
+      parameter do
+        key :paramType, :body
+        key :name, :name
+        key :description, 'The name of the format'
+        key :required, true
+      end
+      parameter do
+        key :name, :format
+        key :in, :body
+        key :description, 'Format to be created'
+        key :required, true
+        schema do
+          key :'$ref', :FormatInput
+        end
+      end
+      extend Api::SwaggerResponses
+      success_schema(model: 'Format')
+    end
+  end
+
+  def create
+    format = FormatSerializer.new(
+      Format.new
+    ).from_hash(params, user_options: { current_user: current_user })
```

Comment:
I think this code should be using the `FormatInput` binding to parse the input and to generate errors if that input is malformed or has bad content.  There's good information in the `FormatInput` schema and we should use it to our advantage.  

`FormatSerializer` as a roar decorator is kind of a duplicate of the binding with a built-in way to stuff data into an AR instance.  I'm not opposed to having `FormatSerializer` class but potentially it could just be the code that takes the binding and creates an AR instance (without Roar - tho I'm not opposed to discussing continuing to use Roar).  

If we keep `FormatSerializer`, I think it should be in the V1 namespace, as it depends on the specifics of this version of the API.  It could be what helps protect you from the model/api version problem I talked about earlier.

Another option is to do the serialization inline, which isn't terrible, e.g. https://github.com/a15k/interactions-api/blob/master/app/controllers/api/v1/flags_controller.rb#L39-L53

Tho I kind of like having a separate class like you.  Maybe I'll add a helper class that wraps the back and forth between the bindings and my models.  Will ponder.